src/Nethermind/Nethermind.Core/Authentication/JwtAuthentication.cs

```diff
@@ -175,6 +194,43 @@ public bool Authenticate(string? token)
         return _timestamper.UnixTime.SecondsLong < exp;
     }
 
+    private static string GetDefaultFilePath()
+    {
+        if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+        {
+            string? xdgDataDir = Environment.GetEnvironmentVariable("XDG_DATA_HOME");
+
+            if (!string.IsNullOrEmpty(xdgDataDir))
+                return Path.Combine(xdgDataDir, "ethereum", "engine", "jwt.hex");
+
+            string? homeDir = Environment.GetEnvironmentVariable("HOME");
+
+            if (string.IsNullOrEmpty(homeDir))
+                throw new Exception("HOME environment variable is not set");
```

Comment:
We also need to find better exception type here

Reply:
My candidates would be either `InvalidOperationException` or `System.Configuration.ConfigurationException`